plugins/woocommerce/includes/rest-api/Controllers/Version4/Products/class-wc-rest-products-v4-controller.php

```
@@ -0,0 +1,2180 @@
+
+				if ( ! wp_attachment_is_image( $attachment_id ) ) {
+					/* translators: %s: image ID */
+					throw new WC_REST_Exception( 'woocommerce_product_invalid_image_id', sprintf( __( '#%s is an invalid image ID.', 'woocommerce' ), $attachment_id ), 400 ); // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+				}
+
+				if ( $is_new_upload && $attachment_id > 0 ) {
+					// Tracking this for rollback purposes.
+					$this->processed_attachment_ids_for_request[] = $attachment_id;
+				}
+
+				$featured_image = $product->get_image_id();
```

PHPMD (2.15.0)

Avoid unused local variables such as '$featured_image'. (Unused Code Rules)

(UnusedLocalVariable)